tests/chainer_tests/links_tests/connection_tests/test_convolution_2d.py

```diff
@@ -305,17 +304,50 @@ def test_forward(self, backend_config):
             (self.height - self.kernel_height + 1) // self.strides_height,
             (self.width - self.kernel_width + 1) // self.strides_width)
 
-    def test_forward_invalid_backend(self):
+
+@testing.inject_backend_tests(
+    [
+        'test_forward',
+    ],
+    # CPU tests
+    [{}]
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'use_cudnn': ['never'],
+    }))
+class TestConvolution2DMemoryLayoutsInvalid(unittest.TestCase,
+                                            TestConvolution2DLayoutParameters):
+
+    def create_link(self):
+        link = links.Convolution2D(
+            self.in_channels,
+            self.out_channels,
+            (self.kernel_height, self.kernel_width),
+            (self.strides_height, self.strides_width))
+        return link
+
+    def create_input_array(self):
+        x_shape = (self.batch, self.height, self.width, self.in_channels)
+        x = cuda.cupy.ones(x_shape, self.dtype)
+        return x
+
+    def test_forward(self, backend_config):
+        if not backend_config.use_cuda:
+            raise unittest.SkipTest(
+                'forward with non-standard layout is only supported with '
+                'cupy arrays.')
         with chainer.using_config('compute_mode', 'cudnn_fast'):
             link = self.create_link()
-        link.to_device('@numpy')
+        link.to_device(backend_config.device)
 
         x = self.create_input_array()
         x = chainer.Variable(x, layout=memory_layouts.CUDNN_CHANNEL_LAST_X)
-        x.to_device('@numpy')
+        x.to_device(backend_config.device)
 
         with pytest.raises(RuntimeError):
-            link(x)
+            with backend_config:
```

Comment:
A nitpick, but lets transpose the `with` statements. `pytest.raises` could catch an exception raised from `backend_config.__enter_()`.